### PR TITLE
Faciliate 3.7 to 3.8 to 3.9 upgrade

### PR DESCRIPTION
**Pull request by sdodson**
Without the exclusion obsoletes trigger an upgrade directly to 3.9.

This can be tested by enabling both the 3.8 and 3.9 repos at the same time and then running playbooks/byo/openshift-cluster/upgrades/v3_9/upgrade_control_plane.yml

I need to squash out some commits but I wanted to leave them in for now as I'm still testing various permutations. I haven't actually got a successful 3.7 containerized install so I haven't had a chance to test that upgrade yet.

### CODE REVIEW

**Comment by sdodson** [comment]
@vrutkovs @mtnbikenc PTAL, still may need some changes but I think this is getting pretty close.

Hopefully we can refactor the upgrade significantly in 3.10 to both reduce the copy/paste across the board but also to make stepwise upgrades generically implemented, I imagine we'll have to do more of this in the future.

File: playbooks/common/openshift-cluster/upgrades/v3_8/upgrade_control_plane.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 - import_playbook: ../init.yml
   vars:
     l_upgrade_no_switch_firewall_hosts: "oo_masters_to_config:oo_etcd_to_config:oo_lb_to_config"
-    l_upgrade_non_node_hosts: "oo_masters_to_config:oo_etcd_to_config:oo_lb_to_config"
```
Comment (vrutkovs) [inline playbooks/common/openshift-cluster/upgrades/v3_8/upgrade_control_plane.yml]: Not sure if this removal is a good idea - its still being used in `playbook/init/facts.yml` and remains in `playbooks/common/openshift-cluster/upgrades/v3_9/upgrade_control_plane.yml`

File: roles/openshift_version/tasks/masters_and_nodes.yml [hunk 2]
```diff
@@ -6,9 +6,12 @@
     include_tasks: check_available_rpms.yml
   - name: Fail if rpm version and docker image version are different
     fail:
-      msg: "OCP rpm version {{ openshift_rpm_version }} is different from OCP image version {{ openshift_version }}"
+      msg: "OCP rpm version {{ rpm_results.results.versions.available_versions.0 }} is different from OCP image version {{ openshift_version }}"
     # Both versions have the same string representation
-    when: rpm_results.results.versions.available_versions.0 != openshift_version
+    when:
+    - openshift_version not in rpm_results.results.versions.available_versions.0
```
Comment (vrutkovs) [inline roles/openshift_version/tasks/masters_and_nodes.yml]: Trailing space - yamllint check fails

File: playbooks/common/openshift-cluster/upgrades/v3_8/upgrade_control_plane.yml (hunk 1 above)
Comment (sdodson) [reply playbooks/common/openshift-cluster/upgrades/v3_8/upgrade_control_plane.yml]: yeah, that was unintentional, thanks for catching it.

**Comment by vrutkovs** [comment]
/lgtm

All issues seem to be fixed, containerized upgrade worked as well

**Pull request merged**